### PR TITLE
Profile screen

### PR DESCRIPTION
**Pull request by noctera**
| Status  | Type  |
| :---: | :---: |
| Ready | Feature |

## Description

Profile screen where you can look at stats and delete your account

Not working:

- change password
- change email

## How Has This Been Tested?

locally

## Screenshots / GIFs (if appropriate):

profile
delete

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking change which cleans up / improves existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [x] I have considered the accessibility of my changes (i.e. did I add proper content descriptions to images, or run my changes with talkback enabled?)
- [x] I have documented my code if needed

### CODE REVIEW

File: src/screens/Profile/Profile.jsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+import TextInput from "../../Components/Form/TextInput/TextInput.jsx";
+import Modal from "../../Components/Modal/Modal.jsx";
+import useSnack from "../../hooks/useSnack.jsx";
+
+import { signOut } from "../../redux/Actions/login.js";
+import { getStats, deleteUser } from "../../utils/api.js";
+
+import "./Profile.scss";
+
+import Table from "../../Components/Table/Table";
```
Comment (luwol03) [inline src/screens/Profile/Profile.jsx:17]: add .jsx

File: src/screens/Profile/Profile.jsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        <h1 style={{ margin: "auto" }}>Hello World</h1>
+      </Modal>
+      <Modal
+        title={t("screens.profile.modals.changeEmail.heading")}
+        open={showEmailModal}
+        onClose={closeEmailModal}
+      >
+        <h1 style={{ margin: "auto" }}>Hello World</h1>
+      </Modal>
+      <Modal
```
Comment (luwol03) [inline src/screens/Profile/Profile.jsx:215]: I think we already have a confirm dialog.

File: src/screens/Profile/Profile.jsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+  const closeDeleteAccountModal = useCallback(() => {
+    setShowDeleteAccountModal(false);
+  }, []);
+
+  useEffect(() => {
+    getProfileStats();
+  });
+
+  //make api call to login
+  const getProfileStats = () => {
```
Comment (luwol03) [inline src/screens/Profile/Profile.jsx:63]: use `useCallback`

File: src/screens/Profile/Profile.jsx [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    } else {
+      setCanSubmitDelete(false);
+    }
+  };
+
+  const handleLogout = useCallback(() => {
+    dispatch(signOut());
+  }, [dispatch]);
+
+  const columns = React.useMemo(
```
Comment (luwol03) [inline src/screens/Profile/Profile.jsx:91]: destructure `useMemo` from React

File: src/screens/Profile/Profile.jsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+  );
+
+  return (
+    <>
+      <div className="profile-screen">
+        <div className="profile-avatar-wrapper">
+          <PersonIcon className="profile-avatar" />
+        </div>
+        <h1 className="profile-username">{username}</h1>
+        <Table columns={columns} data={data} pagination={false} />
```
Comment (luwol03) [inline src/screens/Profile/Profile.jsx:151]: maybe we can move this stats table to a single component so that we can easily reuse this if we need.

**Review by luwol03 — CHANGES_REQUESTED** [review]
image
I think that looks a little too crowded

File: src/screens/Profile/Profile.jsx [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+      >
+        <h1 style={{ margin: "auto" }}>Hello World</h1>
+      </Modal>
+      <Modal
+        title={t("screens.profile.modals.deleteAccount.heading")}
+        open={showDeleteAccountModal}
+        onClose={closeDeleteAccountModal}
+      >
+        <div className="deletion-modal">
+          <TextInput
```
Comment (luwol03) [inline src/screens/Profile/Profile.jsx:221]: maybe we can use the username and/or random number to confirm

File: src/screens/Profile/Profile.jsx [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+    setShowDeleteAccountModal(true);
+  }, []);
+
+  const closeDeleteAccountModal = useCallback(() => {
+    setShowDeleteAccountModal(false);
+  }, []);
+
+  useEffect(() => {
+    getProfileStats();
+  });
```
Comment (luwol03) [inline src/screens/Profile/Profile.jsx:60]: add empty dependencie array to avoid recursion

File: src/screens/Profile/Profile.jsx (hunk 5 above)
Comment (noctera) [reply src/screens/Profile/Profile.jsx:151]: done

**Review by luwol03 — COMMENTED** [review]
Maybe we can also move all dialogs (also confirm dialog) in a separate folder called `src/Dialogs/...`. Or what would you say @zikowang ?

File: src/screens/Profile/Profile.jsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+            value={deleteConfirmation}
+          />
+          <Button
+            block
+            uppercase
+            appearance={"red"}
+            onClick={() => {
+              deleteUser();
+              handleLogout();
+            }}
```
Comment (luwol03) [inline src/screens/Profile/Profile.jsx:159]:
```suggestion
            onClick={async () => {
              await deleteUser();
              handleLogout();
            }}
```

**Comment by zikowang** [comment]
> Maybe we can also move all dialogs (also confirm dialog) into a separate folder called `src/Dialogs/...`. Or what would you say @zikowang ?

I think we just need one customizable dialogue (color / type). We can add a confirmation action like text input in it or add children to it. What do you think?

**Comment by luwol03** [comment]
> > Maybe we can also move all dialogs (also confirm dialog) into a separate folder called `src/Dialogs/...`. Or what would you say @zikowang ?
> 
> I think we just need one customizable dialogue (color / type). We can add a confirmation action like text input or children to it. What do you think?

yes, i would agree with you in that case, since we really don't need the delete account dialog anywhere else.

**Comment by luwol03** [comment]
could you please adjust the Stats table to the changed api Interface in #vocascan/vocascan-server#38

File: src/screens/Profile/Profile.jsx (hunk 8 above)
Comment (noctera) [reply src/screens/Profile/Profile.jsx]: this should be solved with @zikowang fix

```
  const onDelete = useCallback(() => {
    deleteUser()
      .then((response) => {
        dispatch(signOut());
      })
      .catch((e) => {
        console.log(e);
      });
  }, [dispatch]);
  ```

**Comment by noctera** [comment]
> could you please adjust the Stats table to the changed api Interface in [vocascan/vocascan-server#38](https://github.com/vocascan/vocascan-server/pull/38)

done

**Review by luwol03 — APPROVED** [review]

**Review by zikowang — APPROVED** [review]
Please update the screenshots of this PR.

**Comment by noctera** [comment]
> Please update the screenshots of this PR.

done

**Pull request merged**